CMakeLists.txt

```diff
@@ -1,17 +1,17 @@
-cmake_minimum_required(VERSION 3.12.0 FATAL_ERROR)
-
-option(IRODS_ENABLE_430_COMPATIBILITY "Enables compatibility with iRODS 4.3.0." YES) # Default to YES for community.
+cmake_minimum_required(VERSION 3.11.0 FATAL_ERROR)
 
 find_package(IRODS REQUIRED CONFIG)
 
+if (NOT ${IRODS_VERSION} STREQUAL "4.2.11" AND NOT ${IRODS_VERSION} STREQUAL "4.2.12")
+  message(FATAL_ERROR "Incompatible iRODS development package. Requires 4.2.11 or 4.2.12. Found ${IRODS_VERSION}.")
+endif()
```

Comment:
@SwooshyCueb @alanking @trel Any thoughts on approach/message?

I think this is good enough for now, but open to better solutions.

Reply:
seems sufficient to me.   but I don't know what demons we're antagonizing...

Reply:
Tried to compile this branch against 4.3.0 dev pkg and got the following.
```bash
$ cmake -GNinja -DIRODS_ENABLE_430_COMPATIBILITY=NO ../irods_api_plugin_genquery2/
CMake Error at CMakeLists.txt:6 (message):                                                                                                    
  Incompatible iRODS development package.  Requires 4.2.11 or 4.2.12.  Found                                                                  
  4.3.0.                                                                                                                                      
                                                                                                                                              
                                                                                                                                              
-- Configuring incomplete, errors occurred!                                                                                                   
```

I think we're good, but I'm going to verify against 4.2.10 just to be sure.

If all goes well, I'm going to add the pound.

Reply:
Blocked building against 4.2.10 perfectly.
```bash
root@4708174dac52:/_build# cmake -GNinja ../irods_api_plugin_genquery2/
CMake Error at CMakeLists.txt:6 (message):
  Incompatible iRODS development package.  Requires 4.2.11 or 4.2.12.  Found
  4.2.10.


-- Configuring incomplete, errors occurred!
```

Going to tweak the message slightly so everything fits on one line.